github/set_output.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2022 SLSA Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"errors"
+	"os"
+)
+
+// SetOutput writes a name value pair to a file located at GITHUB_OUTPUT.
+func SetOutput(name, value string) error {
+	if filename := os.Getenv("GITHUB_OUTPUT"); filename != "" {
+		f, err := os.OpenFile(filename, os.O_APPEND|os.O_WRONLY, 0o755)
```

Comment:
nit: It's hard to say what the appropriate  #mode should be for the newly created file. The Github Actions toolkit uses the `fs.appendFileSync` default of `0o666` but I doubt they thought about it much.
https://github.com/actions/toolkit/blob/b36e70495fbee083eb20f600eafa9091d832577d/packages/core/src/file-command.ts#L22
https://nodejs.org/api/fs.html#fsappendfilesyncpath-data-options

```suggestion
		f, err := os.OpenFile(filename, os.O_APPEND|os.O_WRONLY, 0o666)
```